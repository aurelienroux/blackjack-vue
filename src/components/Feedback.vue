<template>
  <div class="feedback">
    <p v-if="loading">Loading...</p>
    <p v-if="!gamePlaying">{{ feedback }}</p>
  </div>
</template>

<script>
export default {
  name: 'Feedback',
  props: ['loading', 'gamePlaying', 'feedback']
}
</script>

<style lang="scss">
.feedback {
  align-items: center;
  display: flex;
  flex: 1;
  font-size: 1.25rem;
  justify-content: center;
  min-height: 4.5rem;
}
</style>
