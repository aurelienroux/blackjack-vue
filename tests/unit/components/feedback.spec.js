import { shallowMount } from '@vue/test-utils'
import Feedback from '@/components/Feedback.vue'

function createLocalWrapper(loading = false, gamePlaying = false, feedback = '') {
  return shallowMount(Feedback, {
    propsData: {
      loading,
      gamePlaying,
      feedback
    }
  })
}

describe('Feedback component', () => {
  describe('default state', () => {
    it('renders', () => {
      const wrapper = createLocalWrapper()
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('loading', () => {
    it('renders', () => {
      const wrapper = createLocalWrapper(true)
      expect(wrapper).toMatchSnapshot()
      expect(
        wrapper
          .findAll('p')
          .at(0)
          .text()
      ).toMatch('Loading...')
    })
  })

  describe('game over', () => {
    it('renders', () => {
      const feedbackText = 'feedback text'
      const wrapper = createLocalWrapper(false, false, feedbackText)
      expect(wrapper).toMatchSnapshot()
      expect(wrapper.find('p').text()).toMatch(feedbackText)
    })
  })
})
