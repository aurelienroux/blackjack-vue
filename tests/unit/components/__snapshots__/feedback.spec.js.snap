// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback component default state renders 1`] = `
<div class="feedback">
  <!---->
  <p></p>
</div>
`;

exports[`Feedback component game over renders 1`] = `
<div class="feedback">
  <!---->
  <p>feedback text</p>
</div>
`;

exports[`Feedback component loading renders 1`] = `
<div class="feedback">
  <p>Loading...</p>
  <p></p>
</div>
`;
